### PR TITLE
extend FeePolicy definition

### PR DESCRIPTION
**Pull request by paweljakubas**
# Issue Number

#982

# Overview

- [x] I have extended the definition of FeePolicy
- [x] I have changed Jormungandr binary accordingly
- [x] I have updated core unit testing layer
- [x] I have updated jormungandr unit testing layer
- [x] I have check if updating Sqlite fee policy part is needed (no, as toText is used in PersistField instance definition)

# Comments

### CODE REVIEW

File: lib/core/src/Cardano/Wallet/Primitive/Types.hs [hunk 1]
```diff
@@ -699,26 +699,29 @@ data TransactionInfo = TransactionInfo
...
 -- inputs + outputs.
 --
--- @a@ and @b@ are constant coefficients.
-data FeePolicy =
-    LinearFee (Quantity "lovelace" Double) (Quantity "lovelace/x" Double)
+-- @a@, @b@ and @c@ are constant coefficients.
+data FeePolicy = LinearFee
+    (Quantity "lovelace" Double)
+    (Quantity "lovelace/x" Double)
+    (Quantity "lovelace" Double)
```
Comment (Anviking) [inline lib/core/src/Cardano/Wallet/Primitive/Types.hs]: How about having the units be
- `lovelace`
- `lovelace/byte`
- `lovelace/certificate`
?

File: lib/core/src/Cardano/Wallet/Primitive/Types.hs [hunk 2]
```diff
@@ -699,26 +699,29 @@ data TransactionInfo = TransactionInfo
...
-        [a, b] | T.takeEnd 1 b == "x" -> LinearFee
+        [a, b, c] | T.takeEnd 1 b == "x" -> LinearFee
             <$> fmap Quantity (fromText a)
             <*> fmap Quantity (fromText (T.dropEnd 1 b))
+            <*> fmap Quantity (fromText c)
         _ ->
             Left $ TextDecodingError
                 "Unable to decode FeePolicy: \
-                \Linear equation not in expected format: a + bx"
+                \Linear equation not in expected format: a + bx + c"
```
Comment (Anviking) [inline lib/core/src/Cardano/Wallet/Primitive/Types.hs]: `c` is not a constant here. I presume we need to multiply with some new variable `y` or `certs`.

I wonder where this FromText is used actually…

**Review by Anviking — COMMENTED** [review]
Neat!

File: lib/core/src/Cardano/Wallet/Primitive/Types.hs (hunk 1 above)
Comment (paweljakubas) [reply lib/core/src/Cardano/Wallet/Primitive/Types.hs]: good idea -> changed

File: lib/core/src/Cardano/Wallet/Primitive/Types.hs (hunk 2 above)
Comment (paweljakubas) [reply lib/core/src/Cardano/Wallet/Primitive/Types.hs]: it is used for example in PersistentField instance https://github.com/input-output-hk/cardano-wallet/blob/master/lib/core/src/Cardano/Wallet/DB/Sqlite/Types.hs#L151

File: lib/core/src/Cardano/Wallet/Primitive/Types.hs [hunk 3]
```diff
@@ -699,26 +699,30 @@ data TransactionInfo = TransactionInfo
...
 -- inputs + outputs.
 --
--- @a@ and @b@ are constant coefficients.
-data FeePolicy =
-    LinearFee (Quantity "lovelace" Double) (Quantity "lovelace/x" Double)
+-- @a@, @b@ and @c@ are constant coefficients.
+data FeePolicy = LinearFee
+    (Quantity "lovelace" Double)
+    (Quantity "lovelace/byte" Double)
+    (Quantity "lovelace/certificate" Double)
```
Comment (KtorZ) [inline lib/core/src/Cardano/Wallet/Primitive/Types.hs:706]: :+1:

Comment (KtorZ) [reply lib/core/src/Cardano/Wallet/Primitive/Types.hs:706]: (although, the second value is so-to-speak a `lovelace/number-of-input+output` :/ ... but it captures the same idea.

**Review by KtorZ — APPROVED** [review]

**Pull request merged**